### ISSUE TITLE
C#: Constructor is malformed if last property is readOnly: true

### ISSUE THREAD

**Issue by ggood** [swagger-api/swagger-codegen#2795]
Swagger-codegen 2.1.6. With the OpenAPI spec in this Gist:

https://gist.github.com/ggood/099933cdf7aba3cab4867abd500d1b25

the generated c# constructor won't compile, due to a missing argument:

```
public Foo(string Bar = null, )
        {
            this.Bar = Bar;
            
        }
```

This happens when the last property in a definition is readOnly.

**Comment by wing328** [comment]
@ggood I believe I've addressed this in the latest master. I wonder if you can pull the latest master and give it a try.

**Comment by jimschubert** [comment]
We should probably include a unit test if this is fixed.

I noticed code for required properties last night which I thought might result in this malformed constructor, but I didn't have time to verify my assumption. If this isn't yet fixed, I can grab it early next week.

**Comment by wing328** [comment]
@jimschubert totally agree about adding a test case to cover issues moving forward. I added this before: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore-with-fake-endpoints-models-for-testing.yaml#L807

Let me know that looks good from your perspective (I welcome better and more tests to cover the same issue)

**Comment by jimschubert** [comment]
@wing328 I think it'd be a great idea to add a sample type to `petstore-with-fake-endpoints-models-for-testing.yaml` to verify compilation across all languages. I was thinking of a simple unit test case that compiles the yaml definition from this issue and verifies the string output for the class definition to prevent regression. This would be an additional safety against unintentional changes to `petstore-with-fake-endpoints-models-for-testing.yaml` which would invalidate the test (no longer generating code that would cause the compilation issue).

**Comment by wing328** [comment]
@ggood may I know if you've a chance to try the latest master to see if the issue still persists?

**Comment by ggood** [comment]
I just did a git pull, mvn clean install, and re-ran the generator on my sample. The generated constructor still looks wrong:

```
public Foo(string Bar = null, )
        {


                        this.Bar = Bar;

        }
```

Full constructed class here:

https://gist.github.com/ggood/93a39701368e98e5d59faed56f499857

**Comment by wing328** [comment]
@ggood Thanks for checking. I'll take another look today.

**Comment by wing328** [comment]
@ggood it's my fault that my major [PR](https://github.com/swagger-api/swagger-codegen/pull/2508) to add better enum support for csharp, php, etc removed part of fix by https://github.com/swagger-api/swagger-codegen/pull/2670.

I've added back the fix via #2818.

I'll be more careful rebasing PRs moving forward. Sorry for the inconvenience caused by this.

@jimschubert I misread "snake_case" as the last property but it's not. I've added another property "123Number" (readonly) as the last property and could repeat the issue before the fix so we've this issue covered moving forward. Sorry for the oversight.
